aips/aip-x.md

```diff
@@ -0,0 +1,100 @@
+
+- Developers can now easily implement multisig account creation for any existing account type, not just those with Ed25519 and MultiEd25519 schemes.
+- Users with existing accounts using any authentication scheme can now transition to multisig accounts, which was previously limited.
+- This change enhances the usability and adoption of multisig accounts in the Aptos ecosystem, opening up new possibilities for account management and security across all account types.
+
+## Alternative solutions
+
+The currently existing multisig account creation functions, `create_with_existing_account` and `create_with_existing_account_and_revoke_auth_key`, require a signature of a struct as proof. However, the verification of this signature only works for Ed25519 and MultiEd25519 authentication schemes, excluding accounts with newer schemes like Multi-key or Keyless.
+
+One alternative solution could be to upgrade the on-chain signature verification logic to support all signature types. However, this approach doesn't solve the usability issue of requiring users to sign a struct, which remains cumbersome regardless of the authentication scheme.
+
+This proposal offers a more user-friendly approach by introducing new private entry functions that work with a signer, providing a simpler migration path for all account types, regardless of their authentication scheme.
+
+
+## Specification and Implementation Details
+
+This proposal introduces two new entry functions in the `multisig_account` module:
+
+1. `create_with_existing_account_call`: Creates a multisig account without revoking the original auth key.
+2. `create_with_existing_account_and_revoke_auth_key_call`: Creates a multisig account and rotates the original auth key to 0x0.
+
+These functions are named following the convention of the `account::rotate_authentication_key_call` function, which similarly takes a signer instead of a signature as proof.
+
+## Reference Implementation
+
+// To be replaced with a pull request on aptos-core
+
+## Testing 
+
+The implementation includes new unit tests in the `multisig_account.move` file:
+
+- `test_create_multisig_account_on_top_of_existing_with_signer`: Tests creating a multisig account on top of an existing account using a signer
+- `test_create_multisig_account_on_top_of_existing_and_revoke_auth_key_with_signer`: Tests creating a multisig account and revoking the auth key
+
+These tests cover the basic functionality of the new functions.
+
+## Risks and Drawbacks
+
+The `create_with_existing_account_and_revoke_auth_key_call` function rotates the auth key to 0x0 and revokes capability offers, which irreversibly changes account control. Developers and users should be well-informed about this consequence.
+
```

Comment:
I would mention as with all of these, the more other components (including resource accounts and objects), it may not be clear necessarily if the rotated key is the only thing that can authorize a signer at the address.

Reply:
could you please explain this in simpler terms? i'm not sure i understood this

Reply:
i got what you meant after reading the implementation review comment. change applied